sim-lib/src/lib.rs

```diff
@@ -478,13 +482,22 @@ async fn consume_events(
                         SimulationOutput::SendPaymentSuccess(payment)
                     }
                     Err(e) => {
-                        log::error!(
-                            "Error while sending payment {} -> {}. Error {}.",
-                            node_id,
-                            dest,
-                            e
-                        );
-                        SimulationOutput::SendPaymentFailure(payment, PaymentResult::default())
+                        log::error!("Error while sending payment {} -> {}", node_id, dest);
+
+                        match e {
+                            LightningError::LndRpcError(_) => {
+                                // TODO: Parse LND rpc status to determine if it's a terminal error or not.
+                                SimulationOutput::TerminalError(SimulationError::LightningError(e))
+                            }
+                            LightningError::ClnRpcError(_) => {
+                                // TODO: Parse CLN rpc status to determine if it's a terminal error or not.
+                                SimulationOutput::TerminalError(SimulationError::LightningError(e))
```

Comment:
This is a bit of a layering violation to me, since in practice the code we'll fill in for the `TODO` will likely be some messy implementation-specific string parsing / error code soup. 

What do you think about defining a `LightningError::TemporaryRPCError` that both implementations return from `send_payment` so that the parsing can live in the implementation? 

Then this handling is just: 
- If `TemporaryRPCError` -> send `SendPaymentFailure`
- else -> `TerminalError`

Reply:
Adopted by defining `LightningError:TerminalError(String)`. Any other error we receive is assumed temporary, and Lightning rpc caller chooses how to proceed